### PR TITLE
Homework CMake in SHM

### PR DESCRIPTION
**Pull request by Marcinati**
@TomaszEichler
@Marcinati
@mateusz-talma
@NeoBauer
@WBigaj

### CODE REVIEW

File: shm/main.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    - name: Make build directory
+      run: mkdir shm/build
+    - name: Compile
+      working-directory: shm/build
+      run: |
+        cmake ..
+        make
+    - name: Run tests
+      working-directory: shm/build
+      run: ctest -V
```
Comment (ziobron) [inline shm/main.yml]: this file looks good, but it should be placed into .github/workflows directory. It doesn't have to be named main.yml. After that change, your project should be compiled and run by GitHub automatically :)

File: shm/test.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+#include <string>
+
+class ShipTest : public ::testing::Test {
+protected:
+    std::string name = "First_Ship";
+    size_t maxCrewOfShip = 30;
+    size_t maxCargoCapacity = 100;
+
+    ShipTest() : shipOne(maxCargoCapacity, maxCrewOfShip, 10, name, 1) {}
```
Comment (ziobron) [inline shm/test.cpp]: 10 and 1 are magic numbers

File: shm/test.cpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+TEST_F(ShipTest, ShouldCountAvailableCargoSpaceWhenShipHasNoCargoToSell) {
+    ASSERT_EQ(shipOne.countAvailableSpace(), shipOne.getCapacity());
+}
+
+TEST_F(ItemTest, ShouldGivePriceOfItem) {
+    auto priceAfterRarity = itemOne.getBasePrice() * static_cast<int>(rarity);
+    ASSERT_EQ(itemOne.getPrice(), priceAfterRarity);
+}
```
Comment (ziobron) [inline shm/test.cpp]: Nice tests. A good practice is to have separated test cpp files per test fixture (a class that inherits from ::testing::Test)
BTW - new line missing ;)

**Review by ziobron — APPROVED** [review]
CMakeLists.txt is all good. The only thing you are missing is to put shm/main.yml into .github/workflows directory. It can be merged with my main.yml or you can rename it. Any yml file in this directory should be run by github.

**Comment by ziobron** [comment]
8 x 2XP for each bullet in Narzędzia#3 homework (2767)
2 x 2XP for timely delivery (1119)

You can gain 2 x 2XP for yml file :) I'm waiting for the delivery

{
  "requirement_ids": [2767, 2767, 2767, 2767, 2767, 2767, 2767, 2767, 1119, 1119],
  "users": ["TomaszEichler", "Marcinati", "mateusz-talma", "NeoBauer", "WBigaj"]
}

**Comment by Marcinati** [comment]
All changes was made. ;)

File: shm/CMakeLists.txt [hunk 4]
```diff
@@ -53,7 +53,8 @@ enable_testing()
 
 add_library(${PROJECT_NAME}-lib STATIC ${SRC})
 add_executable(${PROJECT_NAME} main.cpp)
-add_executable(${PROJECT_NAME}-ut test.cpp)
+add_executable(${PROJECT_NAME}-ut testShip.cpp)
+add_executable(${PROJECT_NAME}-ut testItem.cpp)
```
Comment (ziobron) [inline shm/CMakeLists.txt]: Does it work well? I have never tried adding multiple executables with the same name. Normally it should be like this:
```suggestion
add_executable(${PROJECT_NAME}-ut testShip.cpp testItem.cpp)
```

**Review by ziobron — CHANGES_REQUESTED** [review]
You can change the configuration in shm.yml to build on any branch. Your latest changes were not build. The code does not compile:
```bash
error: implicit conversion
      changes signedness: 'int' to 'unsigned long'
      [-Werror,-Wsign-conversion]
    return basePrice_ * static_cast<int>(rarity_);
```

**Comment by Marcinati** [comment]
It's easier to make PR with right one branch (I wanted to make PR with right files only).
Rest of homework yml (file) is on PR #42

**Pull request closed without merging**